csharp/extractor/Semmle.Extraction.CSharp/Entities/PreprocessorDirectives/IfDirective.cs

```diff
@@ -5,18 +5,37 @@ namespace Semmle.Extraction.CSharp.Entities
 {
     internal class IfDirective : PreprocessorDirective<IfDirectiveTriviaSyntax>, IExpressionParentEntity
     {
-        public IfDirective(Context cx, IfDirectiveTriviaSyntax trivia)
+        private IfDirective(Context cx, IfDirectiveTriviaSyntax trivia)
             : base(cx, trivia)
         {
         }
 
         public bool IsTopLevelParent => true;
 
+        public override void WriteId(EscapingTextWriter trapFile)
+        {
+            trapFile.WriteSubId(Context.CreateLocation(ReportingLocation));
+            trapFile.Write(Symbol.IsActive);
+            trapFile.Write(',');
+            trapFile.Write(Symbol.ConditionValue);
+            trapFile.Write(";trivia");
```

Comment:
Shouldn't `Symbol.BranchTaken` be part of the ID?  Do we need `IsActive` and `ConditionValue` in the ID? (Same goes for `elif`, and I think we would need `BranchTaken` in the ID of `else`.)